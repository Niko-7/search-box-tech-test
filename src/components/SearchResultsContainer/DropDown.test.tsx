/*eslint-disable */
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { DropDown } from ".";
const { axe } = require("jest-axe");
import "jest-axe/extend-expect";

afterEach(() => {
  cleanup();
});

describe("DropDown component", () => {
  it("should render the listbox", () => {
    render(
      <DropDown
        locations={[
          {
            placeType: "A",
            city: "Manchester",
            country: "United Kingdom",
            iata: "MAN",
            name: "Manchester Airport",
            region: "Greater Manchester",
          },
        ]}
      />
    );
    const dropdownList = screen.getByRole("listbox");
    expect(dropdownList).toBeInTheDocument;
  });
  it("should render an option", () => {
    render(
      <DropDown
        locations={[
          {
            placeType: "A",
            city: "Manchester",
            country: "United Kingdom",
            iata: "MAN",
            name: "Manchester Airport",
            region: "Greater Manchester",
          },
        ]}
      />
    );
    const dropdownOption = screen.getByRole("option");
    expect(dropdownOption).toBeInTheDocument;
  });
});

describe("accesibility tests", () => {
  it("should have a dropdown list label", () => {
    render(
      <DropDown
        locations={[
          {
            placeType: "A",
            city: "Manchester",
            country: "United Kingdom",
            iata: "MAN",
            name: "Manchester Airport",
            region: "Greater Manchester",
          },
        ]}
      />
    );
    const dropdownLabel = screen.getByLabelText("Search Result List");
    expect(dropdownLabel).toBeInTheDocument;
  });

  it("should have no accesibility errors on the DropDown component", async () => {
    const { container } = render(
      <DropDown
        locations={[
          {
            placeType: "A",
            city: "Manchester",
            country: "United Kingdom",
            iata: "MAN",
            name: "Manchester Airport",
            region: "Greater Manchester",
          },
        ]}
      />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Snapshot tests", () => {
  it("should match the snapshot of the DropDown component", () => {
    const tree = renderer
      .create(
        <DropDown
          locations={[
            {
              placeType: "A",
              city: "Manchester",
              country: "United Kingdom",
              iata: "MAN",
              name: "Manchester Airport",
              region: "Greater Manchester",
            },
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
