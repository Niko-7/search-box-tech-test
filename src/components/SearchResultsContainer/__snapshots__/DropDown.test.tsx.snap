// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should match the snapshot of the DropDown component 1`] = `
<ol
  aria-label="Search Result List"
  className="search-results"
  role="listbox"
>
  <li
    aria-selected="false"
    className="search-results__item"
    id="pickupLocationResultsItem-0"
    role="option"
  >
    <div
      className="search-results__location-type "
    >
      <span>
        
      </span>
    </div>
    <div
      className="search-result-column"
    >
      <div
        className="search-results__location-name"
      >
        Manchester Airport (MAN)
      </div>
      <div
        className="search-results__location-region"
      >
        Manchester, Greater Manchester, United Kingdom
      </div>
    </div>
  </li>
</ol>
`;
